### DISCUSSION TITLE
Ip address issue and VPN/Tor solutions

### DISCUSSION THREAD

**Discussion by ratbunch** [SeanDaBlack/KelloggBot#46] [General]
Currently, the easiest solution for them to filter applications would just be to filter by IP of the user. If it comes to the point where an engineer is told "Hey, fix this!" that would be the first thing I would do as a software dev at Kellog. I'd filter out anything that has different applications from the same IP address. Your public IP which you are currently exposing when running this script and submitting false information.

I've been working on some Tor integration. But would like some input on whether it is worth my time or not. I think the best is obviously to the just have your own custom VPN rotating on US ips. However, this requires some knowledge for the end user to handle this on their own. Free Proxies seem to be too slow for this with stuff as "heavy" as selenium. I just wanted to open a discussion on this and hear any input on what everyone thinks. Is the goal of this repo just "fun"? Is the code meant for SWDevs to run and protect themselves at their own risk? Or is the goal to make a user friendly "click  a button" script?

**Comment by ghost** [comment]
Tor/Proxy support seems like a smart thing to contribute to this project, at least as an option for "power" users (I am defining this as people who arent just gonna point and click). You could run this script from various VPS's but that not everyone has the funds to spend on that. Free proxies or Tor could be a nice alternative

**Reply by ratbunch** [reply]
Cool. I'll mess around with this in my spare time. Though I may hit a deadend if the current script has issues with the slower connection speeds cause by Tor. This seems to be my first issue. I'll fork this and see what I can come up with. Thanks for the input.

**Comment by cafeclimber** [comment]
I found [this script](https://gist.github.com/Lazza/bbc15561b65c16db8ca8) which uses vpngate (apparently a free vpn?) There's obviously a lot of issues with it, but possibly a starting point
